internal/collector/sync_quota_to_backend.go

```diff
@@ -123,36 +134,70 @@ func (c *Collector) performQuotaSync(ctx context.Context, srv db.ProjectService,
 
 	// collect backend quota values that we want to apply
 	targetQuotasInDB := make(map[liquid.ResourceName]uint64)
+	targetAZQuotasInDB := make(map[liquid.ResourceName]map[liquid.AvailabilityZone]liquid.AZResourceQuotaRequest)
 	needsApply := false
+	azSeparatedNeedsApply := false
+	var azSeparatedResouceIDs []db.ProjectResourceID
 	err := sqlext.ForeachRow(c.DB, quotaSyncSelectQuery, []any{srv.ID}, func(rows *sql.Rows) error {
 		var (
+			resourceID   db.ProjectResourceID
 			resourceName liquid.ResourceName
 			currentQuota *int64
 			targetQuota  uint64
 		)
-		err := rows.Scan(&resourceName, &currentQuota, &targetQuota)
+		err := rows.Scan(&resourceID, &resourceName, &currentQuota, &targetQuota)
 		if err != nil {
 			return err
 		}
 		targetQuotasInDB[resourceName] = targetQuota
 		if currentQuota == nil || *currentQuota < 0 || uint64(*currentQuota) != targetQuota {
 			needsApply = true
 		}
+
+		resInfo := c.Cluster.InfoForResource(srv.Type, resourceName)
+		if resInfo.Topology != liquid.AZSeparatedResourceTopology {
+			return nil
+		}
+		err = sqlext.ForeachRow(c.DB, azQuotaSyncSelectQuery, []any{resourceID}, func(rows *sql.Rows) error {
+			var (
+				availabilityZone liquid.AvailabilityZone
+				currentAZQuota   *int64
+				targetAZQuota    uint64
+			)
+			err := rows.Scan(&availabilityZone, &currentAZQuota, &targetAZQuota)
+			if err != nil {
+				return err
+			}
+			// defense in depth: configured backend_quota for AZ any or unknown are not valid for the azSeparatedQuota topology.
+			if (availabilityZone == liquid.AvailabilityZoneAny || availabilityZone == liquid.AvailabilityZoneUnknown) && currentAZQuota != nil {
+				return fmt.Errorf("detected invalid AZ: %s for resource: %s with topology: %s has backend_quota: %v", availabilityZone, resourceName, resInfo.Topology, currentAZQuota)
+			}
+			azSeparatedResouceIDs = append(azSeparatedResouceIDs, resourceID)
+			targetAZQuotasInDB[resourceName] = make(map[liquid.AvailabilityZone]liquid.AZResourceQuotaRequest)
```

Comment:
This should not overwrite existing values from other AZs:
```suggestion
			if targetAZQuotasInDB[resourceName] == nil {
				targetAZQuotasInDB[resourceName] = make(map[liquid.AvailabilityZone]liquid.AZResourceQuotaRequest)
			}
```